akka-docs/src/main/paradox/includes/actor-context.md

```diff
@@ -0,0 +1,15 @@
+
+<!--- #actor-context-typed-access --->
+## Accessing the ActorContext
+
+If the behavior needs to use the `ActorContext`, for example to spawn child actors, or use
+@scala[`context.self`]@java[`context.getSelf()`], it can be obtained by wrapping construction with `Behaviors.setup`:
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #actor-context }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #actor-context }
+
```

Comment:
We can also mention thread-safety here:

```
Many of the methods in `ActorContext` are not thread-safe and must not be accessed from threads
other than the ordinary actor message processing thread, such as
@scala[`scala.concurrent.Future`]@java[`java.util.concurrent.CompletionStage`] callbacks.
It must not be shared between several actor instances.
```

Reply:
Ah ofc - yes might be good to clarify that explicitly.